hoot-services/src/main/java/hoot/services/controllers/grail/GrailResource.java

```diff
@@ -865,42 +871,61 @@ public Response pullRailsPortToDb(@Context HttpServletRequest request,
             throw new UnavailableException("The Rails port API is offline.");
         }
 
-        GrailParams connectedWaysParams = new GrailParams(params);
-        connectedWaysParams.setInput1(referenceOSMFile.getAbsolutePath());
-        File cropFile = new File(params.getWorkDir(), "crop.osm");
-        connectedWaysParams.setOutput(cropFile.getAbsolutePath());
-        // Do an invert crop of this data to get nodes outside bounds
-        workflow.add(grailCommandFactory.build(jobId, connectedWaysParams, "info", InvertCropCommand.class, this.getClass()));
-
-        //read node ids
-        //pull connected ways
-        //pull entire ways
-        //remove cropfile
-        workflow.add(getConnectedWaysApiCommand(jobId, connectedWaysParams));
-
-        // merge OOB connected ways osm files and add 'hoot:change:exclude:delete' tag to each
-        GrailParams mergeOobWaysParams = new GrailParams(params);
-        File mergeOobWaysFile = new File(params.getWorkDir(), "oobways.osm");
-        mergeOobWaysParams.setOutput(mergeOobWaysFile.getAbsolutePath());
-        // Map<String, String> opts = new HashMap<>();
-        // opts.put("convert.ops", "hoot::SetTagValueVisitor");
-        // opts.put("set.tag.value.visitor.element.criteria", "hoot::WayCriterion");
-        // opts.put("set.tag.value.visitor.keys", "hoot:change:exclude:delete");
-        // opts.put("set.tag.value.visitor.values", "yes");
-        // mergeOobWaysParams.setAdvancedOptions(opts);
-        mergeOobWaysParams.setInput1("\\d+\\.osm"); //this is the file filter
-        workflow.add(grailCommandFactory.build(jobId, mergeOobWaysParams, "info", MergeOsmFilesCommand.class, this.getClass()));
-
-        // merge OOB connected ways merge file and the reference osm file
-        GrailParams mergeRefParams = new GrailParams(params);
-        File mergeRefFile = new File(params.getWorkDir(), "merge.osm");
-        mergeRefParams.setInput1("(" + mergeOobWaysFile.getName() + "|" + referenceOSMFile.getName() + ")"); //this is the file filter
-        mergeRefParams.setOutput(mergeRefFile.getAbsolutePath());
-        workflow.add(grailCommandFactory.build(jobId, mergeRefParams, "info", MergeOsmFilesCommand.class, this.getClass()));
+        File ingestFile;
+        // private overpass query result file should handle getting the connected ways so just use that as the ingest file
+        if (isPrivateOverpassActive()) {
+            File connectedWaysOSMFile = new File(params.getWorkDir(), "connectedWays.osm");
+            if (connectedWaysOSMFile.exists()) { connectedWaysOSMFile.delete(); }
+
+            GrailParams connectedWaysParams = new GrailParams(params);
+            connectedWaysParams.setOutput(connectedWaysOSMFile.getAbsolutePath());
+            connectedWaysParams.setCustomQuery(PullApiCommand.connectedWaysQuery());
```

Comment:
Could we even combine the connected ways query with the reference data query (so it would only be a single request to overpass api to get all data in bbox plus connected ways)?  Or would custom filters make that too challenging?

Reply:
I was running into a duplicate node issue, but may have found the option i should be using now